netket/nn/fast_rnn.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2022 The NetKet Authors - All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from flax import linen as nn
+from flax.linen.dtypes import promote_dtype
+
+from jax import numpy as jnp
+from jax.nn.initializers import zeros
+
+from netket.nn.rnn import GRULayer1D, LSTMLayer, RNNLayer
+from netket.utils.types import Array
+
+
+class FastRNNLayer(RNNLayer):
+    """
+    Base class for recurrent neural network layers with fast sampling.
+
+    See :class:`netket.nn.FastMaskedConv1D` for a brief explanation of fast autoregressive sampling.
+    """
```

Comment:
also here, please copy paste a small block explaining why those are `fast` / what it means/ what is the cache / what they assume